07_deploy_masters.sh

```diff
@@ -50,14 +50,18 @@ popd
 
 echo "Master nodes active"
 
-NUM_LEASES=$(sudo virsh net-dhcp-leases baremetal | grep master | wc -l)
-while [ "$NUM_LEASES" -ne 3 ]; do
+# Wait for master nodes to have a DNS record
+NUM_MASTERS=$(dig +short SRV @api.${CLUSTER_DOMAIN} _etcd-server-ssl._tcp.${CLUSTER_DOMAIN} | wc -l)
```

Comment:
@mcornea , you send the DNS query to API VIP (i.e 192.168.111.5), while the DNS VIP is (192.168.111.2).
I guess it should be fine because all masters should have DNS server, and it listens on the DNS Port on all IP/interfaces.
Maybe it worth to add a comment

Reply:
@yboaron Thanks, I updated the comment. The reason I choose querying the API hostname(pointing to the API vip) is that the external DNS server should have a record for the API but not for DNS. Looking at the CI job logs[1] we can see that it worked:

```bash
Mar 27 22:10:18 |  ++ dig +short SRV @api.ostest.test.metalkube.org _etcd-server-ssl._tcp.ostest.test.metalkube.org
Mar 27 22:10:18 |  + NUM_MASTERS=3
Mar 27 22:10:18 |  + '[' 3 -ne 3 ']'
Mar 27 22:10:18 |  ++ awk '{print $NF}'
Mar 27 22:10:18 |  ++ dig +short SRV @api.ostest.test.metalkube.org _etcd-server-ssl._tcp.ostest.test.metalkube.org
Mar 27 22:10:18 |  + for etcd in '$(dig +short SRV @api.${CLUSTER_DOMAIN} _etcd-server-ssl._tcp.${CLUSTER_DOMAIN} | awk '\''{print $NF}'\'')'
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-2.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ head -1
Mar 27 22:10:18 |  + MASTER_NAME=master-2.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-2.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ tail -1
Mar 27 22:10:18 |  + MASTER_IP=192.168.111.22
Mar 27 22:10:18 |  + echo 'You can ssh to master-2.ostest.test.metalkube.org. with core@192.168.111.22'
Mar 27 22:10:18 |  You can ssh to master-2.ostest.test.metalkube.org. with core@192.168.111.22
Mar 27 22:10:18 |  + for etcd in '$(dig +short SRV @api.${CLUSTER_DOMAIN} _etcd-server-ssl._tcp.${CLUSTER_DOMAIN} | awk '\''{print $NF}'\'')'
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-0.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ head -1
Mar 27 22:10:18 |  + MASTER_NAME=master-0.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-0.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ tail -1
Mar 27 22:10:18 |  + MASTER_IP=192.168.111.20
Mar 27 22:10:18 |  + echo 'You can ssh to master-0.ostest.test.metalkube.org. with core@192.168.111.20'
Mar 27 22:10:18 |  You can ssh to master-0.ostest.test.metalkube.org. with core@192.168.111.20
Mar 27 22:10:18 |  + for etcd in '$(dig +short SRV @api.${CLUSTER_DOMAIN} _etcd-server-ssl._tcp.${CLUSTER_DOMAIN} | awk '\''{print $NF}'\'')'
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-1.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ head -1
Mar 27 22:10:18 |  + MASTER_NAME=master-1.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ dig +short CNAME @api.ostest.test.metalkube.org etcd-1.ostest.test.metalkube.org.
Mar 27 22:10:18 |  ++ tail -1
Mar 27 22:10:18 |  + MASTER_IP=192.168.111.21
```

[1] http://10.8.144.11:8080/job/dev-tools/402/consoleFull